iothub/device/samples/how to guides/DeviceReconnectionSample/CustomRetryPolicy.cs

```diff
@@ -11,57 +11,35 @@ namespace Microsoft.Azure.Devices.Client.Samples
     /// An exponential backoff with jitter based retry policy that retries on additional exceptions
     /// not covered by the SDK, because in this application we wish to run as long as possible.
     /// </summary>
-    internal class CustomRetryPolicy : IRetryPolicy
+    internal class CustomRetryPolicy : ExponentialBackoffRetryPolicy
     {
-        private const int MaxRetryCount = int.MaxValue;
-        private const int MaxExponent = 22; // Avoid integer overlow (max of 30) and clamp max wait to just over 1 hour (2^22 = 1.16 hours).
-
-        private readonly Random _rng = new();
-        private readonly object _randLock = new();
         private readonly HashSet<Type> _exceptionsToBeRetried;
-
         private readonly ILogger _logger;
 
         internal CustomRetryPolicy(HashSet<Type> exceptionsToBeRetried, ILogger logger)
+            : base(0, TimeSpan.FromMinutes(1), true)
         {
             _exceptionsToBeRetried = exceptionsToBeRetried ?? new();
             _logger = logger;
         }
 
-        public bool ShouldRetry(int currentRetryCount, Exception lastException, out TimeSpan retryInterval)
+        public override bool ShouldRetry(uint currentRetryCount, Exception lastException, out TimeSpan retryInterval)
         {
-            retryInterval = TimeSpan.Zero;
+            bool shouldRetry;
+
             _logger.LogInformation($"Retry requested #{currentRetryCount} exception [{lastException.GetType()}: {lastException.Message}].");
 
-            if (currentRetryCount > MaxRetryCount)
+            shouldRetry = base.ShouldRetry(currentRetryCount, lastException, out retryInterval);
+            if (!shouldRetry
+                && _exceptionsToBeRetried.Contains(lastException.GetType()))
             {
-                _logger.LogError($"Retry requested #{currentRetryCount} of {MaxRetryCount} so giving up.");
-                return false;
+                retryInterval = TimeSpan.FromSeconds(15);
```

Comment:
is this so that we have a wait time when shouldRetry resolves to false however the exception is marked as "additional" retryable? I think a comment would help here.

Reply:
yes, k